aws/helpers.go

```diff
@@ -0,0 +1,199 @@
+package aws
+
+import (
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// GetNodeHostnames returns the hostnames of the autoscaling group nodes.
+func GetNodeHostnames(autoscalingGroupNodes []*autoscaling.Instance) ([]string, error) {
+	svc := ec2.New(session.New())
+	var instanceHostnames []string
+	for _, node := range autoscalingGroupNodes {
+		resp, err := svc.DescribeInstances(&ec2.DescribeInstancesInput{
+			InstanceIds: []*string{aws.String(*node.InstanceId)},
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to describe ec2 instance")
+		}
+		instanceHostnames = append(instanceHostnames, *resp.Reservations[0].Instances[0].PrivateDnsName)
+	}
+	return instanceHostnames, nil
+}
+
+// GetInstanceID returns the instance ID of a node.
+func GetInstanceID(nodeName string, logger *logrus.Entry) (string, error) {
+	svc := ec2.New(session.New())
+	resp, err := svc.DescribeInstances(&ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("private-dns-name"),
+				Values: []*string{aws.String(nodeName)},
+			},
+		},
+	})
+	if err != nil {
+		return "", errors.Wrap(err, "Failed to describe ec2 instance")
+	}
+
+	if resp.Reservations == nil {
+		logger.Warnf("Instance %s not found, assuming that instance was already deleted", nodeName)
+		return "", nil
+	}
+
+	return *resp.Reservations[0].Instances[0].InstanceId, nil
```

Comment:
Yes make sense an additional check imho.

Reply:
Theoretically line 46 covers this. Because if AWS cannot find the instance it will return a nil for reservations. I think we don't need an extra check for that but 0/5 on that.